src/app/models/AbstractModel.ts

```diff
@@ -60,18 +67,27 @@ export abstract class AbstractModel<Model = Record<string, any>> {
   /**
    * Convert model to JSON
    */
-  public toJSON() {
-    const output = {};
-    this[AbstractModel.attributeKey].forEach((attribute) => {
+  public toJSON(opts?: ToJsonOptions): Partial<this> {
+    const output: Partial<Writeable<this>> = {};
+    let keys: string[];
+    if (!opts) {
+      keys = Object.keys(this).filter((key) => key !== "injector");
```

Comment:
does typescript have a `nameof()` construct for `"injector"`

Reply:
Not sure what you mean

Reply:
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/nameof

https://schneidenbach.gitbooks.io/typescript-cookbook/content/nameof-operator.html